noodles-editor/src/noodles/fields.ts

```diff
@@ -1310,3 +1311,220 @@ export class BezierCurveField extends Field<z.ZodType<BezierCurveData>> {
     return segments
   }
 }
+
+// Helper to detect coordinate field names with various patterns (case-insensitive)
+// Recognizes: lat, latitude, lng, lon, longitude, and any prefixed versions like pickup_lat, dropoff_lng, etc.
+export function findCoordinateFields(obj: Record<string, unknown>): { lng: number; lat: number } | null {
+  if (typeof obj !== 'object' || obj === null) return null
+
+  // Patterns to match longitude and latitude field names (case-insensitive)
+  // Matches: lng, _lng, pickup_lng, dropoff_lon, start_longitude, etc.
+  const lngPatterns = /^(\w*_)?(lng|lon|longitude)$/i
+  const latPatterns = /^(\w*_)?(lat|latitude)$/i
+
+  let lng: number | undefined
+  let lat: number | undefined
+
+  for (const [key, value] of Object.entries(obj)) {
+    if (typeof value === 'number') {
+      if (!lng && lngPatterns.test(key)) {
+        lng = value
+      } else if (!lat && latPatterns.test(key)) {
+        lat = value
+      }
+    }
+  }
+
+  return lng !== undefined && lat !== undefined ? { lng, lat } : null
+}
+
+// Helper to convert a single point to GeoJSON Feature
+export function pointsToFeature(value: unknown): Feature | null {
+  if (!value) return null
+
+  // Already a GeoJSON Feature
+  if (typeof value === 'object' && 'type' in value && value.type === 'Feature') {
+    return value as Feature
+  }
+
+  // Single point as array: [lng, lat]
+  if (Array.isArray(value) && value.length === 2 && typeof value[0] === 'number') {
+    return turf.point([value[0], value[1]], value)
+  }
+
+  // Single point: try to detect coordinate fields with various naming patterns
+  if (typeof value === 'object') {
+    const coords = findCoordinateFields(value as Record<string, unknown>)
+    if (coords) {
+      return turf.point([coords.lng, coords.lat], value)
+    }
+  }
+
+  return null
+}
+
+// Helper to convert various formats to GeoJSON FeatureCollection
+// Converts arrays of points to individual Point features (not MultiPoint)
+export function pointsToFeatureCollection(value: unknown): FeatureCollection | null {
+  if (!value) return null
+
+  // Already a GeoJSON FeatureCollection
+  if (
+    typeof value === 'object' &&
+    'type' in value &&
+    value.type === 'FeatureCollection' &&
+    'features' in value
+  ) {
+    return value as FeatureCollection
+  }
+
+  // Array of features
+  if (
+    Array.isArray(value) &&
+    value.length > 0 &&
+    typeof value[0] === 'object' &&
+    'type' in value[0] &&
+    value[0].type === 'Feature'
```

Comment:
I wonder if the benefits of flat duck typing outweigh the readability of these if statements throughout. Isn't this what zod is typically used for?

One opportunity is when we're validating to a published geojson spec like `isFeature`, `isFeatureCollection`, and `isGeometry`. I'm a bit surprised neither turf nor `geojson` packages include these but its easy to do ourselves.

Something to look at is we check if something is a Feature in four places in two ways in pointsToFeature and pointsToFeatureCollection, and using a different heuristic, in GeometryField and FeatureField.